inc/form.class.php

```diff
@@ -802,7 +802,17 @@ protected function showMyLastForms() {
             }
             echo "</ul>";
             echo '<div align="center">';
-            echo '<a href="form_answer.php?criteria[0][field]=5&criteria[0][searchtype]=equals&criteria[0][value]='.$_SESSION['glpiID'].'">';
+            $criteria = 'criteria[0][field]=5&criteria[0][searchtype]=equals&criteria[0][value]=' . $_SESSION['glpiID'];
+            $critIndex = 1;
+            foreach ($groupIdList as $groupId) {
```

Comment:
since a lot of version of glpi, search engine have a shortcut for this kind of user's group search
value=mygroups for current user.

I don't know if it's applicable here, but i think it's worth it to check.